projects/igniteui-angular/src/lib/grids/filtering/excel-style/excel-style-custom-dialog.component.ts

```diff
@@ -245,6 +245,23 @@ export class IgxExcelStyleCustomDialogComponent implements AfterViewInit {
     }
 
     private createInitialExpressionUIElement() {
+        if (this.expressionsList.length == 1) {
+            const currentExprUI = this.expressionsList.pop();
+            if (currentExprUI.expression.condition.name === this.selectedOperator) {
+                currentExprUI.beforeOperator = FilteringLogic.And;
+                const secondExprUI = new ExpressionUI();
```

Comment:
Then, this code is repeated and can be extracted in a method, I mean the lines that create the secondExpressionUI.